test/common/http/path_utility_fuzz_test.cc

```diff
@@ -0,0 +1,39 @@
+#include "common/http/path_utility.h"
+
+#include "test/common/http/path_utility_fuzz.pb.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/fuzz/utility.h"
+
+namespace Envoy {
+namespace Fuzz {
+namespace {
+DEFINE_PROTO_FUZZER(const test::common::http::PathUtilityTestCase& input) {
+  switch (input.path_utility_selector_case()) {
+  case test::common::http::PathUtilityTestCase::kCanonicalPath: {
+    auto request_headers = fromHeaders<Http::TestRequestHeaderMapImpl>(
+        input.canonical_path().request_headers(), {},
+        {":path"}); // needs to have path header in order to be valid
+    Http::PathUtil::canonicalPath(request_headers);
+    ASSERT(request_headers.get_(":path") != "");
```

Comment:
could try `getPathValue().empty()`. the `get##Name` accessor is O(1), the `get_` will have some more overhead to test whether it is an O(1) header first.